advanced/Scripts/list.sh

```diff
@@ -119,63 +120,64 @@ ProcessDomainList() {
 
 AddDomain() {
     local domain list num
-    domain="$1"
+    # Use printf to escape domain. %q prints the argument in a form that can be reused as shell input
+    domain="$(printf "%q" "$1")"
     list="$2"
 
     # Is the domain in the list we want to add it to?
     num="$(sqlite3 "${gravityDBfile}" "SELECT COUNT(*) FROM ${list} WHERE ${sqlitekey} = \"${domain}\";")"
 
-    if [[ "${num}" -eq 0 ]]; then
-        # Domain not found in the file, add it!
-        if [[ "${verbose}" == true ]]; then
-            echo -e "  ${INFO} Adding ${1} to ${listname}..."
-        fi
-        reload=true
-        # Add it to the list we want to add it to
-        local timestamp
-        timestamp="$(date --utc +'%s')"
-        sqlite3 "${gravityDBfile}" "INSERT INTO ${list} (${sqlitekey},enabled,date_added) VALUES (\"${domain}\",1,${timestamp});"
-    else
-        if [[ "${verbose}" == true ]]; then
-            echo -e "  ${INFO} ${1} already exists in ${listname}, no need to add!"
-        fi
+    if [[ "${num}" -ne 0 ]]; then
+      if [[ "${verbose}" == true ]]; then
+          echo -e "  ${INFO} ${1} already exists in ${listname}, no need to add!"
+      fi
+      return
     fi
+
+    # Domain not found in the table, add it!
+    if [[ "${verbose}" == true ]]; then
+        echo -e "  ${INFO} Adding ${1} to ${listname}..."
+    fi
+    reload=true
+    # Add it to the list we want to add it to
+    local timestamp
+    timestamp="$(date --utc +'%s')"
+    # Insert only domain and date_added here. The enabled fields will be filled
+    # with its default value is true.
+    sqlite3 "${gravityDBfile}" "INSERT INTO ${list} (${sqlitekey},date_added) VALUES (\"${domain}\",${timestamp});"
```

Comment:
Now that the default date added is in the form we want, we don't have to insert it here.